### DISCUSSION TITLE
[Feature]: support Google's Motion Photos

### DISCUSSION THREAD

**Discussion by figadore** [immich-app/immich#1586] [Feature Request]
### Feature detail

Google's "Top Shot" or "Motion Photos" feature embeds a tiny .mp4 file in the jpgs, usually when the file name ends with `.MP.jpg`

It would be great if immich could handle these, using one of the following methods

1. Extracting the mp4 to its own file (in some location or other)
2. Stripping the mp4, reducing the stored file size
3. Being able to play the mp4 when the jpg is viewed

From an [issue in the photoprism repo](https://github.com/photoprism/photoprism/issues/1739), there's a (probably naive) method of getting the mp4:
```
srcfile=image.MP.jpg
offset=$(grep -F --byte-offset --only-matching --text ftyp "$srcfile")
offset=${offset%:*}
dd if=$srcfile of=$srcfile.mp4 bs=$((offset-4)) skip=1
```

### Platform

Server

**Comment by mammo0** [comment]
Duplicate of  #1677?

**Reply by figadore** [reply]
> Duplicate of #1677?

Yes, it appears to be a duplicate, at least for the Android portion, not .heic files

**Comment by benmccann** [comment]
Added in https://github.com/immich-app/immich/pull/950

**Comment by Peronia** [comment]
Hi, I used the Android app version 1.122.0 build.169 (and a server version 1.121.0) and tried to view a local saved "Top Shot" and it is displayed as a normal photo. Did I miss something to get it to work in Immich?
